src/ConstructiveSolidGeometry/SurfacePrimitives/ConeMantle.jl

```diff
@@ -321,3 +321,48 @@ end
 #         return distance_to_line(CartesianPoint(pt), LineSegment(c, _φNear(pcy.φ, φMin, φMax)))
 #     end
 # end
+
+
+function distance_to_line_segment(point::AbstractCoordinatePoint{T}, seg::Tuple{AbstractCoordinatePoint{T},AbstractCoordinatePoint{T}})::T where {T}
+    point = CartesianPoint(point)
+    seg = (CartesianPoint(seg[1]), CartesianPoint(seg[2]))
+    v12 = normalize(CartesianVector{T}(seg[2] - seg[1]))
+    v_point_1 = CartesianVector{T}(point - seg[1])
+    proj_on_v12 = dot(v12,v_point_1)
+    if geom_round(proj_on_v12) ≤ T(0)
+        return norm(seg[1] - point)
+    else
+        v_point_2 = CartesianVector{T}(point - seg[2])
+        if geom_round(dot(v12,v_point_2)) ≥ T(0)
+            return norm(seg[2] - point)
+        else
+            return sqrt(abs(dot(v_point_1,v_point_1) - proj_on_v12^2))
+        end
+    end
+end
+
+@inline _left_linear_interval(z::Real) = -z
+@inline _right_linear_interval(z::Real) = z
+get_r_limits(c::ConeMantle{T, T, <:Any, <:Any}) where {T} = (T(c.r), T(c.r))
+get_r_limits(c::ConeMantle{T, <:Tuple, <:Any, <:Any}) where {T} = c.r
+get_z_limits(c::ConeMantle{T}) where {T} = (c.origin[3]-c.hZ, c.origin[3]+c.hZ)
```

Comment:
This only works, if the `ConeMantle` is neither rotated, nor translated in `x` or `y`.